README.md

```diff
@@ -38,105 +52,212 @@
+````
+├── src
+│   └── org
+│       └── centos
+│           └── pipeline
+│               └── PipelineUtils.groovy
+└── vars
+    └── pipelineUtils.groovy
+````
 
-We assume that there will be tests available to execute otherwise there is nothing besides building, composing, and configuring an Openshift cluster to validate the OStrees and images.  There is currently tests available for many of the Atomic components as well as the Kubernetes e2e conformance tests.
+### CI the CI-Pipeline - Inception
 
-The CI-Pipeline can be defined using Jenkins Job Builder short-term and Jenkins Pipeline (Jenkinsfile) long-term.  This will be the Jenkins 2.0 Pipeline that is already integrated in Openshift [(Openshift Pipelines Deep Dive)](https://blog.openshift.com/openshift-3-3-pipelines-deep-dive/).
+Since we can easily integrate into github then we can use our own pipeline to validate our stage containers, pipeline code, and shared libraries.  
+ 1. We inspect the change.log in the pipeline using the [Jenkins declaritive pipeline DSL](https://github.com/CentOS-PaaS-SIG/ci-pipeline/blob/master/JenkinsfileStageTrigger).
+ 2. If stage containers are touched then they are built in Openshift and validate
+ 3. If the stage container passes validation it is tagged stable and promoted to production in the pipeline
+ 4. If a container and/or pipeline code and/or shared libraries are changed the change is run through th pipeline for validation
 
-We are dependent on the CentOS CI infrastructure for Openshift, Jenkins resources, test resources (bare metal machines, VMs, containers), datahub, and the fedmsg hub.
+#### Example of CI the CI-Pipeline
 
-## Infrastructure and Tools
+![ci-the-ci-pipeline](stage-pipeline-trigger.gif) 
 
-The CI-Pipeline will use CentOS CI for Jenkins master and slave resources.  We plan to move these into an Openshift environment and back our Jenkins with persistent volumes.  Initially all jobs will be defined using [Jenkins Job Builder](https://docs.openstack.org/infra/jenkins-job-builder/), but the team plans in the long-term to move these to a [Jenkins Pipeline](https://jenkins.io/doc/book/pipeline/) using the Jenkinsfile declarative format.  We will be using the [Jenkins JMS plugin](https://wiki.jenkins-ci.org/display/JENKINS/JMS+Messaging+Plugin) to listen and publish messages on fedmsg.  There will be other plugins used such as the email ext and groovy, but most will be part of the core of Jenkins 2.0
+### Pipeline References
 
-The pipeline will use [Linch-pin](https://github.com/CentOS-PaaS-SIG/linch-pin) for the provisioning of resources.  Linch-pin allows us to provision bare metal and VMs in many infrastructures such as AWS, Openstack, GCE, Duffy, libvirt,  and Beaker.
+Currently documentation, Jenkins master/slave containers, container definitions - s2i/docker, pipeline shared libraries, tools, etc. are located [here](https://github.com/CentOS-PaaS-SIG/ci-pipeline).
```

Comment:
Is [here] hyperlinked? Not sure if my browser is just messed up

Reply:
It is supposed to be maybe I need to make it a better statement